### PR TITLE
Upgrade to version 7.2.0

### PR DESCRIPTION
**Pull request by PetrHeinz**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Upgrade demoshop to Shopsys Framework 7.2.0
|New feature| Yes 
|BC breaks| What's considered a BC break in demoshop? 
|Fixes issues| ...
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

- using [general upgrade instructions](https://github.com/shopsys/shopsys/blob/v7.2.0/UPGRADE.md) and [the upgrade guide for 7.2.0](https://github.com/shopsys/shopsys/blob/v7.2.0/docs/upgrade/UPGRADE-v7.2.0.md)
- demoshop doesn't use kubernetes or cloud infrastructure, those upgrades are skipped

### CODE REVIEW

File: src/Shopsys/ShopBundle/Controller/Front/ErrorController.php [hunk 1]
```diff
@@ -159,6 +163,33 @@ private function getPrettyExceptionResponse(Exception $exception)
...
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    private function createUnableToResolveDomainResponse(Request $request): Response
+    {
+        $url = $request->getSchemeAndHttpHost() . $request->getBasePath();
+        $content = sprintf("You are trying to access an unknown domain '%s'.", $url);
+
+        $overwriteDomainUrl = $this->getParameter('overwrite_domain_url');
+        if ($overwriteDomainUrl !== null) {
+            $content .= sprintf(" Parameter 'overwrite_domain_url' is set to '%s'.", $overwriteDomainUrl);
```
Comment (boris-brtan) [inline src/Shopsys/ShopBundle/Controller/Front/ErrorController.php]: hmmm, this was done not by the instructions, would we create reverse PR into monorepo or fix it in UPGRADE-7.2.0.md ?

File: app/config/packages/snc_redis.yml [hunk 2]
```diff
@@ -5,19 +5,19 @@ snc_redis:
...
             options:
-                prefix: '%env(REDIS_PREFIX)%%build-version%bestselling_products_'
+                prefix: '%env(REDIS_PREFIX)%%build-version%:cache:bestselling_products:'
         doctrine_metadata:
             type: 'phpredis'
             alias: 'doctrine_metadata'
             dsn: 'redis://%redis_host%'
             options:
-                prefix: '%env(REDIS_PREFIX)%%build-version%doctrine_metadata_'
+                prefix: '%env(REDIS_PREFIX)%%build-version%:cache:doctrine_metadata:'
```
Comment (boris-brtan) [inline app/config/packages/snc_redis.yml:14]:
```suggestion
                prefix: '%env(REDIS_PREFIX)%%build-version%:cache:doctrine:metadata:'
```

File: app/config/packages/snc_redis.yml [hunk 3]
```diff
@@ -5,19 +5,19 @@ snc_redis:
...
         doctrine_query:
             type: 'phpredis'
             alias: 'doctrine_query'
             dsn: 'redis://%redis_host%'
             options:
-                prefix: '%env(REDIS_PREFIX)%%build-version%doctrine_query_'
+                prefix: '%env(REDIS_PREFIX)%%build-version%:cache:doctrine_query:'
```
Comment (boris-brtan) [inline app/config/packages/snc_redis.yml:20]:
```suggestion
                prefix: '%env(REDIS_PREFIX)%%build-version%:cache:doctrine:query:'
```

File: app/config/packages/snc_redis.yml [hunk 4]
```diff
@@ -29,11 +29,12 @@ snc_redis:
             alias: 'twig'
             dsn: 'redis://%redis_host%'
             options:
-                prefix: '%env(REDIS_PREFIX)%twig_'
+                prefix: '%env(REDIS_PREFIX)%twig:'
         session:
             type: 'phpredis'
             alias: 'session'
             dsn: 'redis://%redis_host%'
+            options:
```
Comment (boris-brtan) [inline app/config/packages/snc_redis.yml:43]: this should not be here

File: app/config/packages/snc_redis.yml [hunk 5]
```diff
@@ -29,11 +29,12 @@ snc_redis:
...
+                prefix: '%env(REDIS_PREFIX)%session:'
     session:
         client: 'session'
-        prefix: '%env(REDIS_PREFIX)%session_'
```
Comment (boris-brtan) [inline app/config/packages/snc_redis.yml:39]:
```suggestion
        prefix: '%env(REDIS_PREFIX)%session:'
```

File: app/config/parameters_common.yml [hunk 6]
```diff
@@ -3,7 +3,7 @@ parameters:
     # Symfony's FrameworkBundle sets throw_at (error_reporting) to 0 in production by default
     debug.error_handler.throw_at: -1
     locale: en
-    build-version: '0000000000000000'
+    build-version: '0000000000000000_%kernel.environment%'
```
Comment (boris-brtan) [inline app/config/parameters_common.yml:7]: will we create some PR in monorepo?

File: CHANGELOG.md [hunk 7]
```diff
@@ -95,6 +95,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
     - added extra error page when current domain cannot be resolved with explicit hint about `overwrite_domain_url` parameter (for TEST environment)
     - using standard redis separators (`:`) in prefixes
     - build version is used in redis prefix of the twig cache and contains the environment (avoids the usage of a wrong cache)
+    - translations are extracted from overwritten templates now
```
Comment (boris-brtan) [inline CHANGELOG.md:98]: are they? won't we also extract them via phing target?

File: app/config/packages/snc_redis.yml [hunk 8]
```diff
@@ -18,6 +18,12 @@ snc_redis:
             dsn: 'redis://%redis_host%'
             options:
                 prefix: '%env(REDIS_PREFIX)%%build-version%:cache:doctrine_query:'
+        framework_annotations:
+            type: 'phpredis'
+            alias: 'framework_annotations'
+            dsn: 'redis://%redis_host%'
+            options:
+                prefix: '%env(REDIS_PREFIX)%%build-version%:cache:framework_annotations:'
```
Comment (boris-brtan) [inline app/config/packages/snc_redis.yml:26]:
```suggestion
                prefix: '%env(REDIS_PREFIX)%%build-version%:cache:framework:annotations:'
```

File: tests/ShopBundle/Functional/Model/Product/Search/ProductSearchExportRepositoryTest.php [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+            'name',
+            'catnum',
+            'partno',
+            'ean',
+            'description',
+            'shortDescription',
+        ];
+
+        if ($productSearchExportRepository instanceof ProductSearchExportWithFilterRepository) {
+            $structure = \array_merge($structure, [
```
Comment (boris-brtan) [inline tests/ShopBundle/Functional/Model/Product/Search/ProductSearchExportRepositoryTest.php:61]: this is strange how `array_keys` method doesn't need global namespace prefix but `array_merge` does

File: src/Shopsys/ShopBundle/Resources/config/services.yml [hunk 10]
```diff
@@ -182,4 +186,3 @@ services:
     Shopsys\ShopBundle\Model\AdvancedSearch\ProductAdvancedSearchConfig: ~
 
     Shopsys\FrameworkBundle\Model\AdvancedSearch\ProductAdvancedSearchConfig: '@Shopsys\ShopBundle\Model\AdvancedSearch\ProductAdvancedSearchConfig'
-
```
Comment (boris-brtan) [inline src/Shopsys/ShopBundle/Resources/config/services.yml:187]:
```suggestion

```

File: src/Shopsys/ShopBundle/Resources/config/services.yml [hunk 11]
```diff
@@ -64,6 +64,10 @@ services:
 
     Shopsys\FrameworkBundle\Model\Product\ProductDataFactoryInterface: '@Shopsys\ShopBundle\Model\Product\ProductDataFactory'
 
+    Shopsys\FrameworkBundle\Model\Product\ProductOnCurrentDomainFacadeInterface: '@Shopsys\FrameworkBundle\Model\Product\ProductOnCurrentDomainElasticFacade'
+
+    Shopsys\FrameworkBundle\Model\Product\Search\Export\ProductSearchExportRepository: '@Shopsys\FrameworkBundle\Model\Product\Search\Export\ProductSearchExportWithFilterRepository'
```
Comment (boris-brtan) [inline src/Shopsys/ShopBundle/Resources/config/services.yml:69]: :smile_cat: shouldn't be here `alias:` :smile_cat:  ???

**Review by boris-brtan — CHANGES_REQUESTED** [review]
Hi @PetrHeinz ,
tHX for the upgrade PR, i just have a few questions and suggestions

File: src/Shopsys/ShopBundle/Resources/config/services.yml (hunk 11 above)
Comment (PetrHeinz) [reply src/Shopsys/ShopBundle/Resources/config/services.yml:69]: I't the same thing (see [Symfony docs on Aliasing](https://symfony.com/doc/current/service_container/alias_private.html#aliasing)) and here the shorthand looks more consistent with the rest of the services.

File: CHANGELOG.md (hunk 7 above)
Comment (PetrHeinz) [reply CHANGELOG.md:98]: Makes sense, I've extracted them in 3c469ca41

File: tests/ShopBundle/Functional/Model/Product/Search/ProductSearchExportRepositoryTest.php (hunk 9 above)
Comment (PetrHeinz) [reply tests/ShopBundle/Functional/Model/Product/Search/ProductSearchExportRepositoryTest.php:61]: We don't have any standards about using functions from the global namespace... This is a copy-pasted file according to our upgrade notes.

Maybe it's something IDE autocompleted for the original author...

File: src/Shopsys/ShopBundle/Controller/Front/ErrorController.php (hunk 1 above)
Comment (PetrHeinz) [reply src/Shopsys/ShopBundle/Controller/Front/ErrorController.php]: I couldn't follow the exact upgrade instructions as the methods were not present in the class at all. So I took this approach. In my point of view, the message is really important only when `overwrite_domain_url` is set to anything but `null`. It doesn't have a direct connection to the TEST environment...

Do we really need the demoshop to be as close to `shopsys/project-base` as possible? I tried to do the upgrade just as I would my own project built on top of Shopsys Framework.

File: app/config/packages/snc_redis.yml (hunk 4 above)
Comment (PetrHeinz) [reply app/config/packages/snc_redis.yml:43]: It's more consistent when all prefixes are configured directly on the clients imo

File: app/config/packages/snc_redis.yml (hunk 5 above)
Comment (PetrHeinz) [reply app/config/packages/snc_redis.yml:39]: It's more consistent when all prefixes are configured directly on the clients imo

File: app/config/packages/snc_redis.yml (hunk 2 above)
Comment (PetrHeinz) [reply app/config/packages/snc_redis.yml:14]: This leads to a more flat and compact  structure and imo it's easier to orient in:

image

File: app/config/parameters_common.yml (hunk 6 above)
Comment (PetrHeinz) [reply app/config/parameters_common.yml:7]: That might make sense. Here on demoshop I ended up with cached Twig template of categories being displayed in `dev` environment, which was very confusing so I've fixed it.

File: src/Shopsys/ShopBundle/Resources/config/services.yml (hunk 10 above)
Comment (PetrHeinz) [reply src/Shopsys/ShopBundle/Resources/config/services.yml:187]: There were two newlines at the end. It's hard to see on Github web UI

File: src/Shopsys/ShopBundle/Controller/Front/ErrorController.php (hunk 1 above)
Comment (PetrHeinz) [reply src/Shopsys/ShopBundle/Controller/Front/ErrorController.php]: OK, I've changed it to the `shopsys/project-base` implementation in c0525d054

**Review by boris-brtan — APPROVED** [review]

**Pull request merged**